doc/OL_preflight_check.py

```diff
@@ -0,0 +1,301 @@
+        "https://airflow.apache.org/docs/apache-airflow-providers-openlineage/stable/guides/user.html"
+    )
+    return
+
+
+def _debug_missing_transport():
+    if _provider_can_be_used():
+        _is_config_set(provider=True)
+        _is_transport_set()
+    _is_config_set(provider=False)
+    _check_openlineage_yml("openlineage.yml")
+    _check_openlineage_yml("~/.openlineage/openlineage.yml")
+    _check_http_env_vars()
+    raise ValueError("OpenLineage is missing configuration, please refer to the OL setup docs.")
+
+
+def _is_listener_accessible():
+    if _provider_can_be_used():
+        try:
+            from airflow.providers.openlineage.plugins.openlineage import OpenLineageProviderPlugin as plugin
+        except ImportError as e:
+            raise ValueError("OpenLineage provider is not accessible") from e
+    else:
+        try:
+            from openlineage.airflow.plugin import OpenLineagePlugin as plugin
+        except ImportError as e:
+            raise ValueError("OpenLineage is not accessible") from e
+
+    if len(plugin.listeners) == 1:
+        return True
+
+    return False
+
+
+def _is_ol_disabled():
+    if _provider_can_be_used():
+        try:
+            # apache-airflow-providers-openlineage >= 1.7.0
+            from airflow.providers.openlineage.conf import is_disabled
+        except ImportError:
+            # apache-airflow-providers-openlineage < 1.7.0
+            from airflow.providers.openlineage.plugins.openlineage import _is_disabled as is_disabled
+    else:
+        from openlineage.airflow.plugin import _is_disabled as is_disabled
+
+    if is_disabled():
+        if _provider_can_be_used() and conf.getboolean("openlineage", "disabled", fallback=False):
+            raise ValueError("OpenLineage is disabled in airflow.cfg: openlineage.disabled")
+        elif os.getenv("OPENLINEAGE_DISABLED", "false").lower() == "true":
+            raise ValueError(
+                "OpenLineage is disabled due to the environment variable OPENLINEAGE_DISABLED"
+            )
+        raise ValueError(
+            "OpenLineage is disabled because required config/env variables are not set. "
+            "Please refer to "
+            "https://airflow.apache.org/docs/apache-airflow-providers-openlineage/stable/guides/user.html"
+        )
+    return False
+
+
+def _get_transport():
+    if _provider_can_be_used():
+        from airflow.providers.openlineage.plugins.openlineage import OpenLineageProviderPlugin
+        transport = OpenLineageProviderPlugin().listeners[0].adapter.get_or_create_openlineage_client().transport
+    else:
+        from openlineage.airflow.plugin import OpenLineagePlugin
+        transport = (
+            OpenLineagePlugin.listeners[0].adapter.get_or_create_openlineage_client().transport
+        )
+    return transport
+
+def is_ol_accessible_and_enabled():
+    if not _is_listener_accessible():
+        _is_ol_disabled()
+
+    transport = _get_transport()
```

Comment:
```suggestion
        try:
        transport = _get_transport()
    except Exception as e:
        raise ValueError("There was an error when trying to build transport.") from e
```

Maybe just for a clarity - i got the error once and had to debug what it's about :)

Reply:
Done